screens/overviewMap.js

```diff
@@ -182,30 +218,72 @@ class OverviewMap extends PureComponent {
     }
   };
 
-  onDragStart = (event) => {
+  onTempPinDragStart = (event) => {
     this.setState({ movingTemp: true });
     event.persist();
   };
 
-  onDrag = (event) => event.persist();
+  onTempPinDrag = (event) => event.persist();
 
-  onDragEnd = (event) => {
+  onTempPinDragEnd = (event) => {
     const coord = event.nativeEvent.coordinate;
     this.setState({ movingTemp: false, tempLocation: coord });
     event.persist();
   };
 
-  onLocationCancel = () => {
-    this.setState({ tempLocation: null });
+  onPinDragStart(event, isUploaded) {
+    this.setState({ draggingPin: true, canDeletePin: !isUploaded });
+    event.persist();
+  }
+
+  onPinDrag(event) {
+    const { canDeletePin } = this.state;
+    const { position } = event.nativeEvent;
+    event.persist();
+
+    if (position.y < this.screenHeight / 8) {
+      if (position.x < this.screenWidth / 2 || !canDeletePin) {
+        this.setState({ inEdit: true, inDelete: false });
+        return;
+      }
+      this.setState({ inEdit: false, inDelete: true });
+      return;
+    }
+
+    this.setState({ inEdit: false, inDelete: false });
+  }
+
+  onPinDragEnd(event, pinLocation) {
+    const { inEdit, inDelete } = this.state;
+    const { deleteLocalLocation } = this.props;
+
+    if (inEdit) {
+      this.createTempPin({ latitude: pinLocation.latitude, longitude: pinLocation.longitude }, pinLocation.key);
+    } else if (inDelete) {
+      deleteLocalLocation(pinLocation.key);
+    } else {
+      // Is there a better way to get the pin to rerender??? Cause this is actually gross...
+      this.setState(() => ({ editingPinKey: pinLocation.key }), () => this.setState(() => ({ editingPinKey: null })));
+    }
+
+    this.setState(() => ({ draggingPin: false, inEdit: false, inDelete: false, canDeletePin: false }));
+    event.persist();
+  }
+
+  onFinishCreation = () => {
+    this.setState({ tempLocation: null, editingPinKey: null });
   };
 
   // eslint-disable-next-line arrow-body-style
   getCreationMaxHeight = () => {
     return /^(pt|fr)/.test(this.props.locale) ? 320 : 280;
   };
 
-  createTempPin = (coord) => {
-    this.setState({ tempLocation: coord });
+  screenWidth = Dimensions.get('window').width * (Platform.OS === 'ios' ? 1 : Dimensions.get('window').scale);
+  screenHeight = Dimensions.get('window').height * (Platform.OS === 'ios' ? 1 : Dimensions.get('window').scale);
+
+  createTempPin = (coord, editingPinKey) => {
+    this.setState(() => ({ tempLocation: coord, editingPinKey }));
```

Comment:
Style: I have since learned it is better to use the following. It updates just those keys whereas what you are doing will overwrite alot more.

```suggestion
     this.setState({ tempLocation: coord, editingPinKey });
```